### PR TITLE
Add start/stop fastscroll callback listener

### PR DESCRIPTION
**Pull request by nicholasrout**
A simple interface to listen to start/stop events of the FastScroller would be useful. My personal case involves hiding a FAB while fast scrolling, so that the bubble is always visible.

### CODE REVIEW

File: example/src/main/java/com/l4digital/fastscroll/example/ExampleActivity.java [hunk 1]
```diff
@@ -41,6 +43,17 @@ protected void onCreate(Bundle savedInstanceState) {
         FastScrollRecyclerView recyclerView = (FastScrollRecyclerView) findViewById(R.id.recycler_view);
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
         recyclerView.setAdapter(new ExampleAdapter());
+        recyclerView.setOnFastScrollStateChangeListener(new OnFastScrollStateChangeListener() {
```
Comment (randr0id) [inline example/src/main/java/com/l4digital/fastscroll/example/ExampleActivity.java]: Can you remove the state change listener from the Example? The example should be kept minimal to demonstrate the simplest implementation of the library.

File: fastscroll/src/main/java/com/l4digital/fastscroll/FastScrollRecyclerView.java [hunk 2]
```diff
@@ -116,6 +116,15 @@ public void setBubbleTextColor(@ColorInt int color) {
         mFastScroller.setBubbleTextColor(color);
     }
 
+    /**
+     * Set the fast scroll state change listener.
+     *
+     * @param onFastScrollStateChangeListener The interface that will listen to fastscroll state change events
+     */
+    public void setOnFastScrollStateChangeListener(OnFastScrollStateChangeListener onFastScrollStateChangeListener) {
```
Comment (randr0id) [inline fastscroll/src/main/java/com/l4digital/fastscroll/FastScrollRecyclerView.java]: The interface name is fine, but I prefer the variable names to not include "on". Can you refactor this variable to `fastScrollStateChangeListener`?

File: fastscroll/src/main/java/com/l4digital/fastscroll/FastScroller.java [hunk 3]
```diff
@@ -73,6 +73,8 @@
     private Drawable mHandleImage;
     private Drawable mTrackImage;
 
+    private OnFastScrollStateChangeListener mOnFastScrollStateChangeListener;
```
Comment (randr0id) [inline fastscroll/src/main/java/com/l4digital/fastscroll/FastScroller.java]: Can you refactor to `mFastScrollStateChangeListener`?

File: fastscroll/src/main/java/com/l4digital/fastscroll/FastScroller.java [hunk 4]
```diff
@@ -274,6 +276,15 @@ public void setBubbleTextColor(@ColorInt int color) {
         mBubbleView.setTextColor(color);
     }
 
+    /**
+     * Set the fast scroll state change listener.
+     *
+     * @param onFastScrollStateChangeListener The interface that will listen to fastscroll state change events
+     */
+    public void setOnFastScrollStateChangeListener(OnFastScrollStateChangeListener onFastScrollStateChangeListener) {
```
Comment (randr0id) [inline fastscroll/src/main/java/com/l4digital/fastscroll/FastScroller.java]: Can you refactor this variable to `fastScrollStateChangeListener`?

**Comment by nicholasrout** [comment]
All requested changes implemented 👍

**Pull request merged**